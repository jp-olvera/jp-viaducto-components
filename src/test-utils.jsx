import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
