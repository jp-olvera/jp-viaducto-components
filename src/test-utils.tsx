import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
