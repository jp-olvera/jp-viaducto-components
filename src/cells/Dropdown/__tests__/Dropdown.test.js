import React from 'react';
import { render, screen, fireEvent } from '../../../test-utils';
import '@testing-library/jest-dom/extend-expect';
import { Dropdown } from '..';

jest.mock('../sorting.svg', () => '');

describe('<Dropdown/>', () => {
  test('should render an focusable activator to toggle the dropdown', () => {
    const dropdown = render(<Dropdown options={['1', '2', '3']} />);
    expect(dropdown.getByTestId('dropdown-activator')).toBeInTheDocument();
    const activator = dropdown.getByTestId('dropdown-activator');
    activator.focus();
    expect(activator).toHaveFocus();
  });
  test('should not display popup', () => {
    const { getByTestId } = render(<Dropdown options={['1', '2', '3']} />);
    expect(getByTestId('dropdown-itemList')).not.toBeVisible();
  });
  test('should display popup', () => {
    const dropdown = render(<Dropdown options={['1', '2', '3']} />);
    const activator = dropdown.getByTestId('dropdown-activator');
    activator.click();
    expect(dropdown.getByTestId('dropdown-itemList')).toBeVisible();
    expect(screen.getByText('1')).toBeVisible();
  });
});
