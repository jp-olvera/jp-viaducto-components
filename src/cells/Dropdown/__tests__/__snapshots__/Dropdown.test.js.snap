// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> should click item 1`] = `undefined`;

exports[`<Dropdown/> should render with default props and border left none 1`] = `
<div>
  <div
    class="StyledDropdown__Wrapper-sc-1btfu08-0 jnKrhx"
    height="2.488rem"
  >
    <button
      aria-controls="dropdown1"
      aria-haspopup="true"
      aria-selected="true"
      class="StyledDropdown__Activator-sc-1btfu08-1 jVWbcI"
      data-testid="dropdown-activator"
      id="dropdown-activator"
      type="button"
    >
      <div
        class="StyledHideable-sc-6qajv2-0 dyQeGA"
      >
        <span
          class="activator-text"
        >
          Buscar por...
        </span>
      </div>
      <img
        alt=""
        class="activator-icon"
      />
    </button>
    <div
      aria-label="Configuraciones"
      class="StyledDropdown__ItemsContainer-sc-1btfu08-2 beJxQA"
      data-cy="dropdown-itemList"
      data-testid="dropdown-itemList"
      id="dropdown1"
    />
  </div>
</div>
`;

exports[`<Dropdown/> should render with default props and borders 1`] = `
<div>
  <div
    class="StyledDropdown__Wrapper-sc-1btfu08-0 jnKrhx"
    height="2.488rem"
  >
    <button
      aria-controls="dropdown1"
      aria-haspopup="true"
      aria-selected="true"
      class="StyledDropdown__Activator-sc-1btfu08-1 gPTejF"
      data-testid="dropdown-activator"
      id="dropdown-activator"
      type="button"
    >
      <div
        class="StyledHideable-sc-6qajv2-0 dyQeGA"
      >
        <span
          class="activator-text"
        >
          Buscar por...
        </span>
      </div>
      <img
        alt=""
        class="activator-icon"
      />
    </button>
    <div
      aria-label="Configuraciones"
      class="StyledDropdown__ItemsContainer-sc-1btfu08-2 beJxQA"
      data-cy="dropdown-itemList"
      data-testid="dropdown-itemList"
      id="dropdown1"
    />
  </div>
</div>
`;
